src/Adapter/Product/CommandHandler/AddProductToPackHandler.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use Pack;
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\AddProductToPackCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\AddProductToPackHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductPackingException;
+use PrestaShopException;
+
+final class AddProductToPackHandler extends AbstractProductHandler implements AddProductToPackHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(AddProductToPackCommand $command): void
+    {
+        // Pack will have the id of product which is being edited
+        $pack = $this->getProduct($command->getPackId());
```

Comment:
You should that `$pack` is a pack itself no?
Or maybe the first time you adda product it's not a Pack yet?

Reply:
> You should that `$pack` is a pack itself no?
> Or maybe the first time you adda product it's not a Pack yet?

Any product can become a pack. There is an assertion later to make sure we are not adding a pack into a pack, but nothing more..
Legacy used to assert if product type is a pack, but that was only a form assertion, command doesn't have anything to do with that. (as we've discussed-technically we can just decide if its a pack in read model by checking if it has pack ralations in db.